testnet/stacks-node/src/tests/signer.rs

```diff
@@ -200,142 +335,183 @@ fn test_stackerdb_dkg() {
-                    && schnorr_proof.is_some()
-                {
-                    break;
-                }
+    info!("------------------------- Test Setup -------------------------");
+    let mut signer_test = SignerTest::new(5, 5);
+
+    // First run DKG in order to sign the block that arrives from the miners following a nakamoto block production
+    // TODO: remove this forcibly running DKG once we have casting of the vote automagically happening during epoch 2.5
+    info!("signer_runloop: spawn send commands to do dkg");
+    signer_test
+        .coordinator_cmd_sender
+        .send(RunLoopCommand::Dkg)
+        .expect("failed to send Dkg command");
+    let mut aggregate_public_key = None;
+    let recv = signer_test
+        .result_receivers
+        .last()
+        .expect("Failed to get coordinator recv");
+    let results = recv.recv().expect("failed to recv results");
+    for result in results {
+        match result {
+            OperationResult::Dkg(point) => {
+                info!("Received aggregate_group_key {point}");
+                aggregate_public_key = Some(point);
+                break;
+            }
+            _ => {
+                panic!("Received Unexpected result");
             }
         }
-        let elapsed = now.elapsed();
-        info!("DKG and Sign Time Elapsed: {:.2?}", elapsed);
     }
-    // Stop the signers
-    for signer in running_signers {
-        assert!(signer.stop().is_none());
+    let aggregate_public_key = aggregate_public_key.expect("Failed to get aggregate public key");
+
+    let (vrfs_submitted, commits_submitted) = (
+        signer_test.running_nodes.vrfs_submitted.clone(),
+        signer_test.running_nodes.commits_submitted.clone(),
+    );
+
+    info!("Mining a Nakamoto tenure...");
+
+    // first block wakes up the run loop, wait until a key registration has been submitted.
+    next_block_and(
+        &mut signer_test.running_nodes.btc_regtest_controller,
+        60,
+        || {
+            let vrf_count = vrfs_submitted.load(Ordering::SeqCst);
+            Ok(vrf_count >= 1)
+        },
+    )
+    .unwrap();
+
+    // second block should confirm the VRF register, wait until a block commit is submitted
+    next_block_and(
+        &mut signer_test.running_nodes.btc_regtest_controller,
+        60,
+        || {
+            let commits_count = commits_submitted.load(Ordering::SeqCst);
+            Ok(commits_count >= 1)
+        },
+    )
+    .unwrap();
+
+    // Mine 1 nakamoto tenure
+    next_block_and_mine_commit(
+        &mut signer_test.running_nodes.btc_regtest_controller,
+        60,
+        &signer_test.running_nodes.coord_channel,
+        &commits_submitted,
+    )
+    .unwrap();
+
+    info!("------------------------- Test Block Processed -------------------------");
+    //Wait for the block to show up in the test observer
+    let validate_responses = test_observer::get_proposal_responses();
```

Comment:
Should this loop and timeout, e.g.,

```rust
let t_start = Instant::now();
while test_observer::get_proposal_responses().is_empty() {
     assert!(t_start.elapsed() < TIMEOUT, "Timed out while waiting for block proposal event");
     thread::sleep(Duration::from_secs(1));
}
```

Otherwise, the test might have a race condition.

Reply:
Done! :)